src/Microsoft.OData.Core/JsonLight/JsonLightODataAnnotationWriter.cs

```diff
@@ -115,6 +136,81 @@ public void WriteInstanceAnnotationName(string annotationName)
             jsonWriter.WriteInstanceAnnotationName(SimplifyODataAnnotationName(annotationName));
         }
 
+        /// <summary>
+        /// Asynchronously writes the odata.type instance annotation with the specified type name.
+        /// </summary>
+        /// <param name="typeName">The type name to write.</param>
+        /// <param name="writeRawValue">Whether to write the raw typeName without removing/adding prefix 'Edm.'/'#'.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        public async Task WriteODataTypeInstanceAnnotationAsync(string typeName, bool writeRawValue = false)
+        {
+            this.AssertAsynchronous();
+            Debug.Assert(typeName != null, "typeName != null");
+
+            // "@odata.type": "#typename"
+            await WriteInstanceAnnotationNameAsync(ODataAnnotationNames.ODataType)
+                .ConfigureAwait(false);
+
+            if (writeRawValue)
+            {
+                await asyncJsonWriter.WriteValueAsync(typeName).ConfigureAwait(false);
+            }
+            else
+            {
+                await asyncJsonWriter.WriteValueAsync(WriterUtils.PrefixTypeNameForWriting(typeName, odataVersion))
+                    .ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Asynchronously writes the odata.type property annotation for the specified property with the specified type name.
+        /// </summary>
+        /// <param name="propertyName">The name of the property for which to write the odata.type annotation.</param>
+        /// <param name="typeName">The type name to write.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        public async Task WriteODataTypePropertyAnnotationAsync(string propertyName, string typeName)
+        {
+            this.AssertAsynchronous();
+            Debug.Assert(!string.IsNullOrEmpty(propertyName), "!string.IsNullOrEmpty(propertyName)");
+            Debug.Assert(typeName != null, "typeName != null");
+
+            // "<propertyName>@odata.type": #"typename"
+            await WritePropertyAnnotationNameAsync(propertyName, ODataAnnotationNames.ODataType)
+                .ConfigureAwait(false);
+            await asyncJsonWriter.WriteValueAsync(WriterUtils.PrefixTypeNameForWriting(typeName, odataVersion))
+                .ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Asynchronously writes a JSON property name which represents a property annotation.
+        /// </summary>
+        /// <param name="propertyName">The name of the property to annotate.</param>
+        /// <param name="annotationName">The name of the annotation to write.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        public Task WritePropertyAnnotationNameAsync(string propertyName, string annotationName)

```

Comment:
>ublic Task [](start = 9, length = 10)

why no async ?

Reply:
@xuzhg For performance reasons, we made the decision to elide `async` and `await` where a method is a pass-through. See Mike's [comment](https://github.com/OData/odata.net/pull/2032#discussion_r600609802) on related PR. More on the subject in this post [https://blog.stephencleary.com/2016/12/eliding-async-await.html](https://blog.stephencleary.com/2016/12/eliding-async-await.html)

Reply:
But is this one a pass-through call? The call to `this.AssertAsynchronous()` has not potential effects/exceptions?

Reply:
I see down below that it just wraps Debug.Assert.